File: leetgo/cpp/LCR 169.di-yi-ge-zhi-chu-xian-yi-ci-de-zi-fu-lcof/solution.cpp
// Created by k at 2024/05/14 17:13
// leetgo: dev
// https://leetcode.cn/problems/di-yi-ge-zhi-chu-xian-yi-ci-de-zi-fu-lcof/

#include <bits/stdc++.h>
#include <map>
#include "LC_IO.h"
using namespace std;

// @lc code=begin

class Solution {
  public:
    char dismantlingAction(string arr)
    {
        map<char, int> m;
        for (int i = 0; i < arr.length(); i++) {
            m[arr[i]] += 1;
        }
        for (int i = 0; i < arr.length(); i++) {
            if (m[arr[i]] == 1) {
                return arr[i];
            }
        }
        return ' ';
    }
};

// @lc code=end

int main()
{
    ios_base::sync_with_stdio(false);
    stringstream out_stream;

    string arr;
    LeetCodeIO::scan(cin, arr);

    Solution *obj = new Solution();
    auto res      = obj->dismantlingAction(arr);
    LeetCodeIO::print(out_stream, res);
    cout << "\noutput: " << out_stream.rdbuf() << endl;

    delete obj;
    return 0;
}

File: leetgo/go/LCR 169.di-yi-ge-zhi-chu-xian-yi-ci-de-zi-fu-lcof/solution.go
// Created by k at 2024/05/14 15:49
// leetgo: dev
// https://leetcode.cn/problems/di-yi-ge-zhi-chu-xian-yi-ci-de-zi-fu-lcof/

package main

import (
	"bufio"
	"fmt"
	"os"

	. "github.com/j178/leetgo/testutils/go"
)

// @lc code=begin

func dismantlingAction(arr string) (ans byte) {
	m := map[rune]int{}
	for _, c := range arr {
		m[c] += 1
	}
	for i, c := range arr {
		if m[c] == 1 {
			return arr[i]
		}
	}
	return ' '
}

// @lc code=end

func main() {
	stdin := bufio.NewReader(os.Stdin)
	arr := Deserialize[string](ReadLine(stdin))
	ans := dismantlingAction(arr)

	fmt.Println("\noutput:", Serialize(ans))
}
